Source/Core/UICommon/ResourcePack/ResourcePack.cpp

```diff
@@ -156,47 +188,40 @@ bool ResourcePack::Install(const std::string& path)
         break;
       }
     }
-
     if (provided_by_other_pack)
       continue;
 
-    if (unzLocateFile(file, ("textures/" + texture).c_str(), 0) != UNZ_OK)
-    {
-      m_error = "Failed to locate texture " + texture;
-      return false;
-    }
-
     const std::string texture_path = path + TEXTURE_PATH + texture;
-    std::string m_full_dir;
-    SplitPath(texture_path, &m_full_dir, nullptr, nullptr);
+    std::string texture_full_dir;
+    if (!SplitPath(texture_path, &texture_full_dir, nullptr, nullptr))
+      continue;
 
-    if (!File::CreateFullPath(m_full_dir))
+    if (!File::CreateFullPath(texture_full_dir))
     {
-      m_error = "Failed to create full path " + m_full_dir;
+      m_error = "Failed to create full path " + texture_full_dir;
       return false;
     }
 
-    unz_file_info texture_info;
-    unzGetCurrentFileInfo(file, &texture_info, nullptr, 0, nullptr, 0, nullptr, 0);
-
-    std::vector<char> data(texture_info.uncompressed_size);
-    if (!Common::ReadFileFromZip(file, &data))
+    const size_t data_size = static_cast<size_t>(texture_info.uncompressed_size);
+    auto data = std::make_unique<u8[]>(data_size);
```

Comment:
Would this be a good candidate for `make_unique_for_overwrite` ?

Reply:
yes but many of the buildbots dont have it

Reply:
Ah that's a shame.  I'll keep that in mind for future PRs.

Reply:
https://en.cppreference.com/w/cpp/compiler_support#cpp20
it seems gcc got it in 11 and clang just doesn't have it :/

Reply:
That makes sense. For reference, which builder do we use clang on?  Yeah clang has been lagging behind..

Reply:
macos and android at least